src/python/pants/backend/python/targets/python_target.py

```diff
@@ -25,6 +27,26 @@ class PythonTarget(Target):
   :API: public
   """
 
+  @classmethod
+  def _suffix_for_synthetic_spec(cls, spec):
+    """Given a spec, append the synthetic resources suffix to it for a consistent mapping."""
+    return spec + '_synthetic_resources'
+
+  @classmethod
+  def create(cls, parse_context, **kwargs):
+    resources = kwargs.get('resources', None) or []
+    if resources:
```

Comment:
Does this preserve the deprecation warning you get when you use an inline resources glob? I don't see it...